morf-core/src/main/java/org/alfasoftware/morf/upgrade/Upgrade.java

```diff
@@ -183,8 +186,17 @@ public UpgradePath findPath(Schema targetSchema, Collection<Class<? extends Upgr
 
     // Get access to the schema we are starting from
     log.info("Reading current schema");
-    Schema sourceSchema = UpgradeHelper.copySourceSchema(connectionResources, dataSource, exceptionRegexes);
-    SqlDialect dialect = connectionResources.sqlDialect();
+    Schema sourceSchema;
+    List<String> schemaConsistencyStatements;
+    try (SchemaResource databaseSchemaResource = connectionResources.openSchemaResource(dataSource)) {
+      sourceSchema = SchemaUtils.copy(databaseSchemaResource, exceptionRegexes);
+      schemaConsistencyStatements = connectionResources.sqlDialect().getSchemaConsistencyStatements(databaseSchemaResource);
+      if (!schemaConsistencyStatements.isEmpty()) {
+        log.warn("Auto-healing statements have been generated (" + schemaConsistencyStatements.size() + " statements in total); this usually implies auto-healing being carried out."
+            + " It this is shown on each subsequent start-up, it can be a symptom of auto-healing failing to achieve an acceptable stable healthful state."
+            + " Examine the upgrade statements (the upgrade script, or the upgrade logs below) to investigate further.");
+      }
+    }

```

Comment:
The schemaConsistencyStatements currently ends here.
It needs to be wired into the UpgradePath and also into GBU.

That is left for a later day though, since it should be somewhat easy to do.

Note: We could also use https://github.com/alfasoftware/morf/pull/321 UpgradeConfiguration, ntroduced separately, before we wire it all in.